homeassistant/components/renson/button.py

```diff
@@ -0,0 +1,76 @@
+"""Sensor data of the Renson ventilation unit."""
+from __future__ import annotations
+
+from _collections_abc import Callable
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator, RensonData
+from .const import DOMAIN
+from .entity import RensonEntity
+
+SYNC_TIME_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="sync_time",
+    entity_category=EntityCategory.CONFIG,
+    translation_key="sync_time",
+)
+
+RESTART_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="restart",
+    device_class=ButtonDeviceClass.RESTART,
+    entity_category=EntityCategory.CONFIG,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renson button platform."""
+
+    data: RensonData = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        RensonButton(
+            SYNC_TIME_BUTTON, data.api, data.coordinator, hass, data.api.sync_time
+        ),
+        RensonButton(
+            RESTART_BUTTON, data.api, data.coordinator, hass, data.api.restart_device
+        ),
+    ]
+
+    async_add_entities(entities)
+
+
+class RensonButton(RensonEntity, ButtonEntity):
+    """Representation of a Renson actions."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        description: ButtonEntityDescription,
+        api: RensonVentilation,
+        coordinator: RensonCoordinator,
+        hass: HomeAssistant,
+        action: Callable,
```

Comment:
Personally I prefer if this is set in the entity description but it's not a stopper